torch/_dynamo/variables/lists.py

```diff
@@ -296,6 +296,23 @@ def call_method(
     ) -> "VariableTracker":
         return super().call_method(tx, name, args, kwargs)
 
+class SetVariable(BaseListVariable):
```

Comment:
Well RIP, there is no SetVariable. This issue became a bit more complicated.

I think for the initial version, you should produce a VariableTracker consistent with whatever the set() constructor would have produced, if you had done a list comprehension and then passed it to set. You should run a small test program to find out what VariableTracker is.

A bigger change is to implement a SetVariable. But sets are not lists, so this implementation is not correct. It would probably be better to model it similar to ConstDictVariable in dict.py

Reply:
for the initial version, do you mean the base call function is
```
def call_method(
        self,
        tx,
        name,
        args: "List[VariableTracker]",
        kwargs: "Set[VariableTracker]",
    ) -> "VariableTracker":
        return super().call_method(tx, name, args, kwargs)
```
I'm not clear about how to run a small test program to find out what VariableTracker is

Reply:
> Well RIP, there is no SetVariable. This issue became a bit more complicated.
> 
> I think for the initial version, you should produce a VariableTracker consistent with whatever the set() constructor would have produced, if you had done a list comprehension and then passed it to set. You should run a small test program to find out what VariableTracker is.
> 
> A bigger change is to implement a SetVariable. But sets are not lists, so this implementation is not correct. It would probably be better to model it similar to ConstDictVariable in dict.py

for the initial version, do you mean the base call function is
```
def call_method(
        self,
        tx,
        name,
        args: "List[VariableTracker]",
        kwargs: "Set[VariableTracker]",
    ) -> "VariableTracker":
        return super().call_method(tx, name, args, kwargs)
```
I'm not clear about how to run a small test program to find out what VariableTracker is

Reply:
It's ConstantVariable apparently.

I found out by

```
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (9f7dd5a4)]$ cat mmm.py 
import torch
@torch.compile()
def fn(x):
    s = set(["lsh", "local"])
    return x + 2

fn(torch.randn(2))
(/home/ezyang/local/a/pytorch-env) [ezyang@devgpu020.ftw1 ~/local/a/pytorch (9f7dd5a4)]$ TORCH_LOGS=+dynamo python mmm.py
```

and then in the logs you can see

```
[2023-04-24 07:27:53,779] torch._dynamo.symbolic_convert: [DEBUG] TRACE BUILD_LIST 2 [BuiltinVariable(set), Constant
Variable(str), ConstantVariable(str)]           
[2023-04-24 07:27:53,779] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 1 [BuiltinVariable(set), ListV
ariable()]                                                
[2023-04-24 07:27:53,779] torch._dynamo.symbolic_convert: [DEBUG] TRACE STORE_FAST s [ConstantVariable(set)] 
```

Reply:
get a nightly, this is new